_posts/2021-02-03-spring-boot-s3.md

```diff
@@ -0,0 +1,424 @@
+
+AWS gives us multiple ways to automatically delete objects from a bucket, however we do use S3's concept of `Object Life Cycle rules`. An object life cycle rule basically contains the information *when* to do *what* with the object:
+
+```java
+// delete files a week after upload
+s3Client
+  .setBucketLifecycleConfiguration(
+    "my-awesome-bucket",
+    new BucketLifecycleConfiguration()
+      .withRules(
+        new BucketLifecycleConfiguration.Rule()
+          .withId("custom-expiration-id")
+          .withFilter(new LifecycleFilter())
+          .withStatus(BucketLifecycleConfiguration.ENABLED)
+          .withExpirationInDays(7)
+      )
+  );
+```
+
+We use the client's `setBucketLifecycleConfiguration` method, given the bucket's name and the desired configuration. The configuration above consists of a single rule, having:
+- an `id` to make the rule uniquely identifiable
+- a default `LifecycleFilter`, so this rule applies to all objects in the bucket
+- a status of being `ENABLED`, so as soon as this rule is created id is effective
+- an expiration of seven days, so after a week the object gets deleted
+
+It shall be noted that the snippet above overrides the old lifecycle configuration. That is ok for our use case but you might want to fetch the existing rules first and upload the combination of old and new rules.
+
+> **Sample Application: Set expiration on Bucket**
+>
+> To use the sample application's UI to set an expiration on a bucket's objects:
+> 1. Navigate to the _Spaces_ section
+> 2. Select _Make Temporary_ on the target Space/Bucket
+> 3. Select _Delete_ on the target object
+> 4. A message should pop up to indicate success
+
+A noteworthy around lifecycle rules is that they are far more versatile, as you can use the filter to only apply the rule to objects with a certain key prefix or carry out other actions like archiving of objects.
+
+## Conclusion
+
+In this article, you've learned the basics of AWS' Simple Storage Service (S3) and how to use Spring Boot and the `Spring Cloud` project to get started with it.
+
+We used S3 to build a custom file-sharing application, that lets you upload & share your files in different ways. But it shall be said, that S3 is way more versatile, often also quoted to be the **backbone of the internet**.
+
+As this is a getting started article, we did not touch other topics like [storage tiers](https://aws.amazon.com/s3/storage-classes/), [object versioning](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Versioning.html) or [static content hosting](https://docs.aws.amazon.com/AmazonS3/latest/dev/WebsiteHosting.html). So I can only recommend you get your hands dirty, and play around with S3!
```

Comment:
Perhaps add a link to the code on github in the conclusion.

Reply:
Added the code the same way as in the _Sample Application_ section